corehq/apps/accounting/models.py

```diff
@@ -1230,6 +1232,13 @@ def send_ending_reminder_email(self):
                     'email': email,
                 })
 
+    def set_billing_account_entry_point(self):
+        no_current_entry_point = self.account.entry_point == EntryPoint.NOT_SET
+        self_serve = self.service_type == SubscriptionType.SELF_SERVICE
+        if (no_current_entry_point and self_serve and not self.is_trial):
+            self.account.entry_point = EntryPoint.SELF_STARTED
+            self.account.save()
```

Comment:
there's a lot of this type of stuff in the billing code and something I'd like to remove/explore more proactively.

in form/case processing we are very aggressive at reducing saves to improve atomicity and reduce db load. in peripheral code like this it's less critical, but still something we should strive for.

sorry, that's not helpful, and I don't have context to say what should be done (if anything) here. the short answer is that we don't have good conventions but should.